### PR TITLE
[BUGFIX] fix incorrect pandas top rows usage

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GE-320/GE-364
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/map_metric.py [hunk 1]
```diff
@@ -553,15 +553,14 @@ def _pandas_column_map_condition_values(
 
     domain_values = df[column_name]
 
+    domain_values = domain_values[boolean_mapped_unexpected_values == True]
```
Comment (cdkini) [inline great_expectations/expectations/metrics/map_metric.py:556]: Can we consolidate this into a single filtered df? Is that too complex to put on a single line?

File: great_expectations/expectations/metrics/map_metric.py [hunk 2]
```diff
@@ -803,12 +796,12 @@ def _pandas_map_condition_rows(
 
     result_format = metric_value_kwargs["result_format"]
 
+    df = df[boolean_mapped_unexpected_values]
+
     if result_format["result_format"] == "COMPLETE":
```
Comment (cdkini) [inline great_expectations/expectations/metrics/map_metric.py:801]: Is there any chance of `result_format` missing the "result_format" key? Should we make this a `get()`?

File: great_expectations/expectations/metrics/map_metric.py [hunk 3]
```diff
@@ -620,24 +619,20 @@ def _pandas_column_map_series_and_domain_values(
...
+
     result_format = metric_value_kwargs["result_format"]
+
     if result_format["result_format"] == "COMPLETE":
         return (
-            list(domain_values[boolean_map_unexpected_values == True]),
-            list(map_series[boolean_map_unexpected_values == True]),
+            list(domain_values),
+            list(map_series),
         )
```
Comment (cdkini) [inline great_expectations/expectations/metrics/map_metric.py:631]: Thoughts on using `df.columns.tolist()` or `df.columns.values.tolist()`? The latter is more performant but I'm just thinking from a readability standpoint. I wasn't sure what `list(df)` did but that might just be from not having used `pandas` in a while.

File: tests/expectations/metrics/test_core.py [hunk 4]
```diff
@@ -563,6 +563,25 @@ def test_map_unique_pd_column_exists():
...
+        },
+    )
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_rows_metric,), metrics=metrics
+    )
+    metrics.update(results)
+
+    assert metrics[unexpected_rows_metric.id]["a"].index == [2]
+    assert metrics[unexpected_rows_metric.id]["a"].values == [3]
+
```
Comment (cdkini) [inline tests/expectations/metrics/test_core.py:584]: What is this test doing? Does it add coverage over this `pandas` issue that was not covered before? Should it be a separate test entirely or are we okay with appending it here?

I don't have the requisite knowledge to definitively say but I thought it was prudent to bring up such questions so you could evaluate.

**Review by cdkini — APPROVED** [review]
I reviewed from a purely Python perspective since my knowledge of the underlying systems is lacking. I see the pandas related issue and see how it's resolved so as long as tests pass, I'll sign off.

Please do take a look at my comments (non-blocking) and let me know your thoughts.

File: tests/expectations/metrics/test_core.py (hunk 4 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_core.py:584]: This tests goes into a metric and the result format that triggers the behavior, where the number of rows returned from the pandas dataframe has to be less than its total number of rows.

**Pull request merged**